Services/Init/classes/class.ilStartUpGUI.php

```diff
@@ -236,6 +236,14 @@ protected function showLoginPage(ilPropertyFormGUI $form = null)
             $tpl->setVariable('LPE', $page_editor_html);
         }
 
+        if(array_key_exists('wdtdel', $this->httpRequest->getQueryParams())) {
```

Comment:
Please factor this out to a new `TermsOfService` class. The footprint in `ilStartupGUI` should be as small as possible.